src/ns/adapters/inbound/event_sub.py

```diff
@@ -31,25 +39,48 @@ class EventSubTranslatorConfig(NotificationEventsConfig):
 class EventSubTranslator(EventSubscriberProtocol):
     """A translator that can consume Notification events"""
 
-    def __init__(self, *, config: EventSubTranslatorConfig, notifier: NotifierPort):
+    def __init__(
+        self,
+        *,
+        config: EventSubTranslatorConfig,
+        notifier: NotifierPort,
+        event_id_dao: EventIdDaoPort,
+    ):
         self.topics_of_interest = [config.notification_topic]
         self.types_of_interest = [config.notification_type]
         self._config = config
         self._notifier = notifier
+        self._event_id_dao = event_id_dao
 
     async def _send_notification(self, *, payload: JsonObject):
         """Validates the schema, then makes a call to the notifier with the payload"""
         validated_payload = get_validated_payload(
             payload=payload, schema=event_schemas.Notification
         )
+
         await self._notifier.send_notification(notification=validated_payload)
 
     async def _consume_validated(
-        self, *, payload: JsonObject, type_: Ascii, topic: Ascii, key: Ascii
+        self,
+        *,
+        payload: JsonObject,
+        type_: Ascii,
+        topic: Ascii,
+        key: Ascii,
+        event_id: UUID,
     ) -> None:
         """Consumes an event"""
-        if (
-            type_ == self._config.notification_type
-            and topic == self._config.notification_topic
-        ):
-            await self._send_notification(payload=payload)
+        # Don't need to check for topic and type because we only subscribe to one topic
+        # and hexkit ensures that the event is of the correct type.
+        with suppress(ResourceNotFoundError):
+            _ = await self._event_id_dao.get_by_id(event_id)
+            log.info("Notification already processed, skipping. Event_id=%s", event_id)
+            return
+
+        # Let the DLQ handle any errors that bubble up
```

Comment:
Not exactly clear on how the DLQ is involved here.
Is there something missing/should this have some error handling here?